elasticsearch/src/main/java/org/apache/zeppelin/elasticsearch/ElasticsearchInterpreter.java

```diff
@@ -0,0 +1,369 @@
+      "" + response.getHits().getTotalHits());
+  }
+
+  /**
+   * Processes a "search" request.
+   * 
+   * @param urlItems Items of the URL
+   * @param data May contains the limit and the JSON of the request
+   * @return Result of the search request, it contains a tab-formatted string of the matching hits
+   */
+  private InterpreterResult processSearch(String[] urlItems, String data) {
+
+    if (urlItems.length > 2) {
+      return new InterpreterResult(InterpreterResult.Code.ERROR,
+                                   "Bad URL (it should be /index1,index2,.../type1,type2,...)");
+    }
+        
+    final SearchResponse response = searchData(urlItems, data);
+
+    return new InterpreterResult(
+      InterpreterResult.Code.SUCCESS,
+      InterpreterResult.Type.TABLE,
+      buildResponseMessage(response.getHits().getHits()));
+  }
+
+  /**
+   * Processes a "index" request.
+   * 
+   * @param urlItems Items of the URL
+   * @param data JSON to be indexed
+   * @return Result of the index request, it contains the id of the document
+   */
+  private InterpreterResult processIndex(String[] urlItems, String data) {
+        
+    if (urlItems.length < 2 || urlItems.length > 3) {
+      return new InterpreterResult(InterpreterResult.Code.ERROR,
+                                   "Bad URL (it should be /index/type or /index/type/id)");
+    }
+        
+    final IndexResponse response = client
+      .prepareIndex(urlItems[0], urlItems[1], urlItems.length == 2 ? null : urlItems[2])
+      .setSource(data)
+      .get();
+
+    return new InterpreterResult(
+      InterpreterResult.Code.SUCCESS,
+      InterpreterResult.Type.TEXT,
+      response.getId());
+  }
+
+  /**
+   * Processes a "delete" request.
+   * 
+   * @param urlItems Items of the URL
+   * @return Result of the delete request, it contains the id of the deleted document
+   */
+  private InterpreterResult processDelete(String[] urlItems) {
+
+    if (urlItems.length != 3 
+        || StringUtils.isEmpty(urlItems[0]) 
+        || StringUtils.isEmpty(urlItems[1]) 
+        || StringUtils.isEmpty(urlItems[2])) {
+      return new InterpreterResult(InterpreterResult.Code.ERROR,
+                                   "Bad URL (it should be /index/type/id)");
+    }
+
+    final DeleteResponse response = client
+      .prepareDelete(urlItems[0], urlItems[1], urlItems[2])
+      .get();
+        
+    if (response.isFound()) {
+      return new InterpreterResult(
+        InterpreterResult.Code.SUCCESS,
+        InterpreterResult.Type.TEXT,
+        response.getId());
+    }
+        
+    return new InterpreterResult(InterpreterResult.Code.ERROR, "Document not found");
+  }
+    
+  private SearchResponse searchData(String[] urlItems, String data) {
+
+    final SearchRequestBuilder reqBuilder = new SearchRequestBuilder(client, SearchAction.INSTANCE);
+        
+    if (urlItems.length >= 1) {
+      reqBuilder.setIndices(StringUtils.split(urlItems[0], ","));
```

Comment:
should we check the result of `StringUtils.split(urlItems[0], ",")` here and below?
is it ok if there is no `,`?